packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,113 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  const cacheFunc = (ttl: number) =>
+    cacheDecorator.cacheMiddleware(ttl, cosmosRPCKeyGenerator);
+
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802'];
+  const completedCodes = [
+    '0803',
+    '0804',
+    '0805',
+    '0803220a0a080000000000000087',
+    '0803220a0a080000000000000100',
+    '0803220a0a080000000000000189',
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
+    let body = req.body;
+    try {
+      body = JSON.parse(req.body);
+    } catch (err) {
+      console.log('Error parsing body: ', err);
+    }
+
+    const params = JSON.stringify(body?.params);
+    const method = body?.method;
+
+    if (/^(status|block|tx)/.test(method)) {
+      // LIVENESS CHECK  or TX Response: do not cache
+      return next();
+    }
+
+    if (/^Query\/(Votes|TallyResult|Pool|Deposits)/.test(params)) {
+      return next(); // LIVE data matters do not cache
+    }
```

Comment:
Maybe a short cache is worth it here too. Performance improvements are probably more important to us than up-to-the-second vote results.

Reply:
made this a 1-block cache